### PR TITLE
Address management actor test

### PR DESCRIPTION
**Pull request by kirtisagarksj**
Address management actor test

### CODE REVIEW

File: actors/user/src/test/java/org/sunbird/user/AddressManagementActorTest.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    } catch (Exception e) { // TODO Auto-generated catch block
+      Assert.fail("Initialization failed");
+    }
+    cassandraOperation = mock(CassandraOperationImpl.class);
+    when(ServiceFactory.getInstance()).thenReturn(cassandraOperation);
+  }
+
+  @Test
+  public void testInsertAddressSuccess() {
+    TestKit probe = new TestKit(system);
```
Comment (amolnin) [inline actors/user/src/test/java/org/sunbird/user/AddressManagementActorTest.java]: again there is a pattern in each -testcase - one common function receiving the 
(String requestObjType, boolean success, Response response)

Comment (kirtisagarksj) [reply actors/user/src/test/java/org/sunbird/user/AddressManagementActorTest.java]: yes so?

Comment (amolnin) [reply actors/user/src/test/java/org/sunbird/user/AddressManagementActorTest.java]: so create such function -> and call it from each test-case -> test-code will be reduced quite a bit

File: actors/user/src/test/java/org/sunbird/user/AddressManagementActorTest.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+        .thenReturn(getSuccessResponse());
+    getAssertResult("updateUserAddress", true);
+  }
+
+  @Test
+  public void testUpdateAddressFailure() {
+    getAssertResult("updateUserAddress", true);
+  }
+
+  private void getAssertResult(String operation, boolean success) {
```
Comment (amolnin) [inline actors/user/src/test/java/org/sunbird/user/AddressManagementActorTest.java]: testScenario

File: actors/user/src/test/java/org/sunbird/user/AddressManagementActorTest.java [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    PowerMockito.mockStatic(org.sunbird.common.models.util.datasecurity.impl.ServiceFactory.class);
+    EncryptionService encryptionService = Mockito.mock(EncryptionService.class);
+    Mockito.when(
+            org.sunbird.common.models.util.datasecurity.impl.ServiceFactory
+                .getEncryptionServiceInstance(null))
+        .thenReturn(encryptionService);
+
+    try {
+      Mockito.when(encryptionService.encryptData(Mockito.anyString())).thenReturn("encrptUserId");
+    } catch (Exception e) { // TODO Auto-generated catch block
```
Comment (amolnin) [inline actors/user/src/test/java/org/sunbird/user/AddressManagementActorTest.java]: remove TODO comment

File: actors/user/src/test/java/org/sunbird/user/AddressManagementActorTest.java [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+      Assert.fail("Initialization failed");
```
Comment (amolnin) [inline actors/user/src/test/java/org/sunbird/user/AddressManagementActorTest.java]: AddressManagementActorTest initialization failed

File: actors/user/src/test/java/org/sunbird/user/AddressManagementActorTest.java [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+  @Test
+  public void testUpdateAddressSuccess() {
+    when(cassandraOperation.deleteRecord(
+            Mockito.anyString(), Mockito.anyString(), Mockito.anyString()))
+        .thenReturn(getSuccessResponse());
+    getAssertResult("updateUserAddress", true);
+  }
+
+  @Test
+  public void testUpdateAddressFailure() {
```
Comment (amolnin) [inline actors/user/src/test/java/org/sunbird/user/AddressManagementActorTest.java]: testUpdateAddressFailureWithoutMandatoryFields

File: actors/user/src/test/java/org/sunbird/user/AddressManagementActorTest.java [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+    subject.tell(getRequestObject(operation, success), probe.getRef());
+    Response res = probe.expectMsgClass(duration("10 second"), Response.class);
+    if (success) {
+      Assert.assertTrue(res != null && res.getResult().get(JsonKey.RESPONSE) == "SUCCESS");
+    } else {
+      Assert.assertTrue(res != null && res.getResult().get(JsonKey.ERROR_MSG) != null);
+    }
+  }
+
+  private Request getRequestObject(String operation, boolean isParamReq) {
```
Comment (amolnin) [inline actors/user/src/test/java/org/sunbird/user/AddressManagementActorTest.java]: isParamReq -> withReqParams

File: actors/user/src/test/java/org/sunbird/user/AddressManagementActorTest.java [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+    reqObj.setOperation(operation);
+    if (isParamReq) {
+      reqObj.put(JsonKey.ADDRESS, getAddressList());
+      reqObj.put(JsonKey.ID, "someId");
+      reqObj.put(JsonKey.CREATED_BY, "createdBy");
+    }
+    return reqObj;
+  }
+
+  private Object getAddressList() {
```
Comment (amolnin) [inline actors/user/src/test/java/org/sunbird/user/AddressManagementActorTest.java]: Return type cannot be object - it should be list

File: actors/user/src/test/java/org/sunbird/user/AddressManagementActorTest.java [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+      Mockito.when(encryptionService.encryptData(Mockito.anyString())).thenReturn("encrptUserId");
+    } catch (Exception e) { // TODO Auto-generated catch block
+      Assert.fail("Initialization failed");
+    }
+    cassandraOperation = mock(CassandraOperationImpl.class);
+    when(ServiceFactory.getInstance()).thenReturn(cassandraOperation);
+  }
+
+  @Test
+  public void testInsertAddressSuccess() {
```
Comment (amolnin) [inline actors/user/src/test/java/org/sunbird/user/AddressManagementActorTest.java]: testInsertUserAddress

File: actors/user/src/test/java/org/sunbird/user/AddressManagementActorTest.java [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+    when(cassandraOperation.insertRecord(
+            Mockito.anyString(), Mockito.anyString(), Mockito.anyMap()))
+        .thenReturn(getSuccessResponse());
+    getAssertResult("insertUserAddress", true);
+  }
+
+  @Test
+  public void testInsertAddressFailure() {
```
Comment (amolnin) [inline actors/user/src/test/java/org/sunbird/user/AddressManagementActorTest.java]: testInserUserAddressFailureWithoutReqParams

File: actors/user/src/test/java/org/sunbird/user/AddressManagementActorTest.java [hunk 10]
```diff
@@ -0,0 +1,133 @@
...
+    getAssertResult("insertUserAddress", false);
+  }
+
+  @Test
+  public void testUpdateAddressSuccess() {
+    when(cassandraOperation.deleteRecord(
+            Mockito.anyString(), Mockito.anyString(), Mockito.anyString()))
+        .thenReturn(getSuccessResponse());
+    getAssertResult("updateUserAddress", true);
```
Comment (amolnin) [inline actors/user/src/test/java/org/sunbird/user/AddressManagementActorTest.java]: Use ActorOperations enum

Comment (kirtisagarksj) [reply actors/user/src/test/java/org/sunbird/user/AddressManagementActorTest.java]: its not defined in ActorOperations.java

**Pull request closed without merging**